easybuild/easyblocks/h/hadoop.py

```diff
@@ -0,0 +1,85 @@
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Hadoop, implemented as an easyblock
+
+@author: Kenneth Hoste (Ghent University)
+"""
+import os
+import shutil
+
+from easybuild.easyblocks.generic.tarball import Tarball
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.run import run_cmd
+
+
+class EB_Hadoop(Tarball):
+    """Support for building/installing Hadoop."""
+
+    @staticmethod
+    def extra_options():
+        """Custom easyconfig parameters for Hadoop."""
+        extra_vars = {
+            'build_native_libs': [False, "Build native libraries", CUSTOM],
+         }
+        return Tarball.extra_options(extra_vars)
+
+    def build_step(self):
+        """Custom build procedure for Hadoop: build native libraries, if requested."""
+        if self.cfg['build_native_libs']:
+            cmd = "mvn package -DskipTests -Dmaven.javadoc.skip -Dtar -Pdist,native"
+            run_cmd(cmd, log_all=True, simple=True, log_ok=True)
+
+    def install_step(self):
+        """Custom install procedure for Hadoop: install-by-copy."""
+        if self.cfg['build_native_libs']:
+            src = os.path.join(self.builddir, 'hadoop-dist', 'target', 'hadoop-%s' % self.version)
+            try:
+                shutil.rmtree(self.installdir)
+                shutil.copytree(src, self.installdir)
+            except OSError, err:
+                self.log.error("Failed to copy contents of %s to install dir %s: %s" % (src, self.installdir, err))
+        else:
+            super(EB_Hadoop, self).install_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for Hadoop."""
+        native_files = []
+        if self.cfg['build_native_libs']:
+            native_files = ['lib/native/libhadoop.so']
+
+        custom_paths = {
+            'files': ['bin/hadoop'] + native_files,
+            'dirs': ['etc', 'libexec'],
+        }
+        super(EB_Hadoop, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_extra(self):
+        """Custom extra module file entries for Hadoop."""
+        txt = super(EB_Hadoop, self).make_module_extra()
+        mapreduce_subdir = os.path.join('share', 'hadoop', 'mapreduce')
+        txt += self.module_generator.prepend_paths('HADOOP_HOME', mapreduce_subdir)
+        txt += self.module_generator.prepend_paths('HADOOP_CONF_DIR', os.path.join(mapreduce_subdir, 'conf'))
```

Comment:
`HADOOP_CONF_DIR` needs to point to the directory where configs are. AFAICS, for Hadoop installs this is usually in the install path but for installs on HPC systems I can't see how this could work correctly since the configs would need to be configured -and EB doesn't really handle that. So they would either not define it and have users provide it or, if they have a site wide install, install it with a config mgmnt tool.

So we can remove the HADOOP_CONF_DIR part.